hbase-server/src/main/java/org/apache/hadoop/hbase/io/MetricsIO.java

```diff
@@ -24,10 +24,12 @@
 @InterfaceAudience.Private
 public class MetricsIO {
 
+  private static volatile MetricsIO instance;
   private final MetricsIOSource source;
   private final MetricsIOWrapper wrapper;
 
-  public MetricsIO(MetricsIOWrapper wrapper) {
+  // visible for testing only
+  MetricsIO(MetricsIOWrapper wrapper) {
```

Comment:
weird here, here is implementing a singleton, but there is still a public constructor...

is it used somewhere? if not, could we just delete it.

Reply:
Yes it is being used in a test case, hence had to keep it and changed scope from public